src/components/AvatarRow/AvatarRow.jsx

```diff
@@ -0,0 +1,168 @@
+import React, { useRef, useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import throttle from 'lodash.throttle'
+import classNames from 'classnames'
+import { noop } from '../../utilities/other'
+import { AvatarRowUI, CounterAvatarUI } from './AvatarRow.css'
+import Avatar from '../Avatar'
+import Tooltip from '../Tooltip'
+import { config } from '../Avatar/Avatar.css'
+import { setupObserver, splitAvatarsArray } from './AvatarRow.utils'
+import { number } from '@storybook/addon-knobs'
+
+const { size: avatarConfigSizes } = config
+
+function AvatarRow({
+  adaptable = true,
+  avatars = [],
+  className,
+  'data-cy': dataCy = 'AvatarRow',
+  extraTooltipProps = {},
+  gap = 2,
+  ieCompatible = false,
+  throttleOnResize = true,
+  throttleWait = 200,
+  ...avatarProps
+}) {
+  const { size: avatarSize, fontSize: avatarFontSize } = avatarConfigSizes[
+    avatarProps.size || 'md'
+  ]
+  const avatarRowRef = useRef(null)
+  const [numberOfItemsOnDisplay, setNumberOfItemsOnDisplay] = useState(
+    avatars.length
+  )
+
+  const throttledOnResize = !ieCompatible
+    ? throttle(onResize, throttleWait)
+    : noop
+  const throttledHandleWindowResize = ieCompatible
+    ? throttle(handleWindowResize, throttleWait)
+    : noop
+
+  useEffect(() => {
+    if (!adaptable || ieCompatible || avatarRowRef.current == null) return
+
+    const avatarRowEl = avatarRowRef.current
+    const resizeObserver = setupObserver(
+      throttleOnResize ? throttledOnResize : onResize
+    )
+
+    resizeObserver.observe(avatarRowEl)
+
+    return () => {
+      if (!adaptable || ieCompatible || avatarRowEl == null) return
+
+      resizeObserver.unobserve(avatarRowEl)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (ieCompatible && adaptable) {
+      window.addEventListener(
+        'resize',
+        throttleOnResize ? throttledHandleWindowResize : handleWindowResize
+      )
+
+      const measures = avatarRowRef.current.getBoundingClientRect()
+      onResize(measures)
+    }
+
+    return () => {
+      ieCompatible &&
+        adaptable &&
+        window.addEventListener(
+          'resize',
+          throttleOnResize ? throttledHandleWindowResize : handleWindowResize
+        )
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  function handleWindowResize() {
+    if (avatarRowRef.current != null) {
+      const measures = avatarRowRef.current.getBoundingClientRect()
+
+      onResize(measures)
+    }
+  }
+
+  function onResize({ width: containerWidth }) {
+    /** Only act if we have more than 1 avatar */
+    if (!containerWidth || avatars.length <= 1) {
+      return
+    }
+
+    const spaceForAllAvatars =
```

Comment:
Could you add a comment to this explaining this calculation? I'm having trouble understanding the `(avatars.length - 1)` part.

Reply:
No problem! Let me know if it makes it clearer

Reply:
Nice thanks!